Display green "Shared" banner after user shares; bring CSS into line with spec

benjiwheeler opened the pull request:
### Resolves:

Resolves https://github.com/LLK/scratch-www/issues/2240

### Changes:

Displays green "Shared" banner on project page after user shares, which persists until user browses away or makes a change.

### Test Coverage:

none

benjiwheeler commented:
Thanks for the suggestions! Take a look.

benjiwheeler commented:
Here's what this looks like now:
<img width="1030" alt="screen shot 2018-11-28 at 6 30 27 pm" src="https://user-images.githubusercontent.com/3431616/49195410-d6f80a80-f33b-11e8-9944-be5f8a7814ff.png">

(Note that new scratchers will see a longer message than this; this is what Scratchers will see:)
<img width="1008" alt="screen shot 2018-11-28 at 6 30 38 pm" src="https://user-images.githubusercontent.com/3431616/49195411-d6f80a80-f33b-11e8-9fef-c851625e7c20.png">

paulkaplan commented:
@benjiwheeler there are some lint errors

benjiwheeler commented:
Arrgh, ok, take a look now!

paulkaplan commented:
@benjiwheeler why were the CSS changes needed? Some of them smell a bit, like having fixed heights on buttons (what if the text wraps?). I thought the banner looked ok with a single line of text with no button, like the "Note: your project is in the trash" one?

paulkaplan commented on src/views/preview/presentation.jsx:
```diff
@@ -128,12 +130,28 @@ const PreviewPresentation = ({
-        />);
+    } else if (canShare) {
+        if (isShared) {
+            if (justShared) {
```
Can't you just add justShared to the isShared if statement (i.e. `if (isShared && justShared) { ...`) to reduce the nesting and allow you to put the helpful comment that is currently at the bottom of this block alongside that condition?

benjiwheeler replied:
No, because the `else` for the unshared state (below) depends only on `isShared` and not on `justShared`. I did go back and forth on whether the way I have it now is better or worse than doing 

```
if (canShare && isShared && justShared) {
    if (isNewScratcher) {
        // ...
    } else {
        // ...
    }
} else if (canShare && !isShared) {
   // ...
}
```

...which is slightly more redundant, but maybe easier to read?

benjiwheeler commented on src/views/preview/project-view.jsx:
```diff
@@ -665,6 +673,7 @@ const mapStateToProps = state => {
             state.permissions.admin === true),
         isLoggedIn: isLoggedIn,
         isAdmin: isAdmin,
+        isNewScratcher: isLoggedIn && state.permissions.new_scratcher,
```
Good point. Yeah, as helpful as these individual flags (and the `can`- flags) are, they can be a little misleading -- like how it's hard to know exactly what `canCreateNew` means, since in a sense, any user can create new at any time.

It could be better to give these more elaborate names, such as `isLoggedInNewScratcher`, but even for me (who prefers verbosity more than most!) that seems like overkill.

So I appreciate your noting this.

benjiwheeler commented:
The CSS changes are to bring the banner in line with the invision specs that I know of.

I changed that `height` to a `min-height` and added a few more mins and maxes.
